### PR TITLE
features: gatekeeper validation tests doc

### PR DESCRIPTION
**Pull request by mmirecki**
Depends on https://github.com/openshift-kni/cnf-features-deploy/pull/592/

### CODE REVIEW

File: cnf-tests/GATEKEEPER.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# Gatekeeper testing
+
+This document describes how to test set up and test Gatekeeper manually.
```
Comment (oribon) [inline cnf-tests/GATEKEEPER.md:3]: can you specify here that the setup is via the operator and maybe link to their README?

File: cnf-tests/GATEKEEPER.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+spec:
+  audit:
+    replicas: 1
+  mutatingWebhook: "Enabled"
+  webhook:
+    failurePolicy: "Ignore"
+EOF
+```
+
+Check if the gatekeeper deployments are running by executing the following command:
```
[the remark this answers is not in the document]
Comment (mmirecki) [reply cnf-tests/GATEKEEPER.md:30]: Yes: gatekeeper-controller-manager and gatekeeper-audit

File: cnf-tests/GATEKEEPER.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+Sample constraint templates and constraints can be found in ` feature-configs/demo/gatekeeper`.
+
+To apply these please run:
+
+```shell
+oc create -f feature-configs/demo/gatekeeper/00_ConstraintTemplates.yaml
+oc create -f feature-configs/demo/gatekeeper/10_Constraints.yaml 
+```
+
+NOTE: please wait a few seconds inbetween the commands to allow gatekeeper to process them.
```
[the remark this answers is not in the document]
Comment (mmirecki) [reply cnf-tests/GATEKEEPER.md:47]: fixed

File: cnf-tests/GATEKEEPER.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  mutatingWebhook: "Enabled"
+  webhook:
+    failurePolicy: "Ignore"
+EOF
+```
+
+Check if the gatekeeper deployments are running by executing the following command:
+
+```shell
+oc get deployment -n gatekeeper-system
```
[the remark this answers is not in the document]
Comment (mmirecki) [reply cnf-tests/GATEKEEPER.md:33]: added

File: cnf-tests/GATEKEEPER.md (hunk 1 above)
Comment (mmirecki) [reply cnf-tests/GATEKEEPER.md:3]: done

File: cnf-tests/GATEKEEPER.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
+# Gatekeeper testing
+
+This document describes how to test set up and test Gatekeeper manually.
+It describes how to install Gatekeeper using the [gatekeeper-operator](https://github.com/gatekeeper/gatekeeper-operator), setting up some basic mutations and how to validatate the muations are applied correctly.
```
Comment (SchSeba) [inline cnf-tests/GATEKEEPER.md:4]: muations -> mutation

Comment (SchSeba) [reply cnf-tests/GATEKEEPER.md:4]: maybe point to the ocp repo?

File: cnf-tests/GATEKEEPER.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+## Gatekeeper setup
+
+The Gatekeeper operator can be installed by running the following command in the root of the cnf repo:
```
Comment (SchSeba) [inline cnf-tests/GATEKEEPER.md:8]: cnf -> cnf-feature-deploy

File: cnf-tests/GATEKEEPER.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+Sample constraint templates and constraints can be found in ` feature-configs/demo/gatekeeper`.
+
+To apply these please run:
+
+```shell
+oc create -f feature-configs/demo/gatekeeper/00_ConstraintTemplates.yaml
+oc create -f feature-configs/demo/gatekeeper/10_Constraints.yaml 
+```
+
+NOTE: please wait a few seconds between the commands to allow gatekeeper to process them.
```
Comment (SchSeba) [inline cnf-tests/GATEKEEPER.md:59]:
```
NOTE: -> *NOTE:*
```

File: cnf-tests/GATEKEEPER.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+Gatekeeper will reject this pod with a message similar to the one below:
+
+```
+Error from server ([deny-master-no-schedule-toleration] Toleration is not allowed for taint {"effect": "NoSchedule", "key": "node-role.kubernetes.io/master", "value": "true"}): error when creating "STDIN": admission webhook "validation.gatekeeper.sh" denied the request: [deny-master-no-schedule-toleration] Toleration is not allowed for taint {"effect": "NoSchedule", "key": "node-role.kubernetes.io/master", "value": "true"}
+```
+
+This pod was rejected by the deny-master-no-schedule-toleration constraint of type k8srestrictspecifictoleration.constraints.gatekeeper.sh
+
+##### Pod with a global taint
+
```
Comment (SchSeba) [inline cnf-tests/GATEKEEPER.md:123]: nit: remove this empty line

**Review by SchSeba — COMMENTED** [review]
just a small nits and we can merge it

File: cnf-tests/GATEKEEPER.md (hunk 5 above)
Comment (mmirecki) [reply cnf-tests/GATEKEEPER.md]: done

File: cnf-tests/GATEKEEPER.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+```shell
+oc create -f feature-configs/demo/gatekeeper/00_ConstraintTemplates.yaml
+oc create -f feature-configs/demo/gatekeeper/10_Constraints.yaml 
+```
+
+NOTE: please wait a few seconds between the commands to allow gatekeeper to process them.
+
+With the constraints in place, create some pods to validate that the constraints are applied properly.
+To restrict the effect of the validation on the cluster, the constraints are only valid for namespaces having a `admission.gatekeeper.sh/tolerations: "enforce"` label.
+Create such a namespace by executing:
```
[the remark this answers is not in the document]
Comment (mmirecki) [reply cnf-tests/GATEKEEPER.md:64]: done

**Pull request merged**